modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/FAPIDCRValidationsTestCase.java

```diff
@@ -0,0 +1,183 @@
+            description = "Check FAPI validations, PPID and SSA during DCR", dataProvider = "dcrConfigProvider")
+    public void validateErrorScenarios(JSONObject requestJSON, String errorCode, String errorMessage) throws Exception {
+
+        HttpPost request = new HttpPost(DCRUtils.getPath(tenant));
+        request.addHeader(HttpHeaders.AUTHORIZATION, DCRUtils.getAuthzHeader(username, password));
+        request.addHeader(HttpHeaders.CONTENT_TYPE, OAuthDCRMConstants.CONTENT_TYPE);
+        StringEntity entity = new StringEntity(requestJSON.toJSONString());
+        request.setEntity(entity);
+        HttpResponse response = client.execute(request);
+
+        assertEquals(response.getStatusLine().getStatusCode(), 400, "Service Provider " +
+                "has not been created successfully");
+        JSONObject errorResponse = DCRUtils.getPayload(response);
+        assertEquals(errorResponse.get("error"), errorCode);
+        assertEquals(errorResponse.get("error_description"), errorMessage);
+    }
+    @Test(alwaysRun = true, groups = "wso2.is", priority = 11,
+            description = "Check FAPI validations, PPID and SSA during DCR", dataProvider = "dcrConfigProvider")
+    public void validateErrorScenariosForDCRUpdate(JSONObject requestJSON, String errorCode, String errorMessage)
+            throws Exception {
+
+        // Create application.
+        HttpPost request = new HttpPost(DCRUtils.getPath(tenant));
+        JSONObject registerRequestJSON = DCRUtils.getRegisterRequestJSON("request6.json");
+
+        request.addHeader(HttpHeaders.AUTHORIZATION, DCRUtils.getAuthzHeader(username, password));
+        request.addHeader(HttpHeaders.CONTENT_TYPE, OAuthDCRMConstants.CONTENT_TYPE);
+        StringEntity entity = new StringEntity(registerRequestJSON.toJSONString());
+        request.setEntity(entity);
+        ObjectMapper mapper = new ObjectMapper();
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 201, "Service Provider " +
+                "has not been created successfully");
+        JSONObject createResponsePayload  = DCRUtils.getPayload(response);
+        client_id = ((JSONObject) createResponsePayload).get("client_id").toString();
+        assertNotNull(client_id, "client_id cannot be null");
+
+        // Check error scenarios for update request.
+        HttpPut updateRequest = new HttpPut(DCRUtils.getPath(tenant) + client_id);
+        request.addHeader(HttpHeaders.AUTHORIZATION, DCRUtils.getAuthzHeader(username, password));
+        request.addHeader(HttpHeaders.CONTENT_TYPE, OAuthDCRMConstants.CONTENT_TYPE);
+        entity = new StringEntity(requestJSON.toJSONString());
+        request.setEntity(entity);
+        response = client.execute(updateRequest);
+
+        assertEquals(response.getStatusLine().getStatusCode(), 400, "Service Provider " +
+                "has not been created successfully");
```

Comment:
Shouldn't the message be something like. `Service Provider should not be created successfully`

Reply:
check other relevant places as well

Reply:
changed